Se agregan los campos faltantes de service.py

cuenta56 opened the pull request:
Se agregan los campos faltantes al services.py con el formato necesario para agregar usuarios desde el REST Framework tanto al dispositivo con un JSON como a la base de datos SQL.

cla1976 committed:
Se agregan los campos faltantes de service.py

cla1976 commented on user_profile_api/models.py:
```diff
@@ -57,6 +64,14 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
     date_created = models.DateTimeField(auto_now=True, editable=False)
     last_updated = models.DateTimeField(auto_now=True, editable=False)
+    dni = models.CharField(max_length=100, unique=True, null=True)
+    id = models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')
```
Cambiar nombre de campo por device_id

cla1976 commented on user_profile_api/models.py:
```diff
@@ -57,6 +64,14 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
+    phone = models.TextField(max_length=100, null=True, blank=True)
+    emergency_phone = models.TextField(max_length=100, null=True)
+    devices_total = models.CharField(max_length=100, choices=DEVICES)
+    begin_time = models.DateTimeField(editable=True, null=True)
+    end_time = models.DateTimeField(editable=True, null=True)
```
Estos 2 campos quitarlos porque no los necesitamos guardar en la DB, solo los usamos para enviar el dato al Dispositivo

cla1976 commented on user_profile_api/serializers.py:
```diff
@@ -28,16 +36,15 @@ class Meta:
         }
 
-    def create(self, validated_data):
+    def create_user(self, validated_data):
```
El nombre del metodo tiene que ser create()

cla1976 commented on user_profile_api/serializers.py:
```diff
@@ -28,16 +36,15 @@ class Meta:
             email=validated_data['email'],
             first_name=validated_data['first_name'],
             last_name=validated_data['last_name'],
-            profile_type=validated_data['profile_type'],
```
Esta linea no hay que borrarla, dejarla como esta

Agregar los demas campos del model: dni, device_id, etc, etc

cla1976 commented on user_profile_api/serializers.py:
```diff
@@ -28,16 +36,15 @@ class Meta:
 
         user.set_password(validated_data['password'])
-        user.save()
+        user.save(using=self._db)
```
Esta linea tambien debe quedar como estaba, sin el parametro using=self._db

cla1976 commented on user_profile_api/services.py:
```diff
@@ -35,22 +35,32 @@ def search_users():
+    full_name = f'{first_name} {last_name}'
+   
+    begin_time = request.validated_data["begin_time"]
+    begin_time_final = begin_time.strftime("%Y-%m-%dT%H:%M:%S")
+    end_time = request.validated_data["end_time"]
+    end_time_final = end_time.strftime("%Y-%m-%dT%H:%M:%S")
```
Agregar un metodo generico dentro de utils.py que devuelva el string "%Y-%m-%dT%H:%M:%S" para poder usarlo en cualquier lado

cla1976 commented on user_profile_api/services.py:
```diff
@@ -61,10 +71,10 @@ def record_user():
+        
+        res = requests.post(full_url, headers=headers, data=payload, auth=HTTPDigestAuth(GATEWAY_USER, GATEWAY_PASSWORD))	
         data = res.json()
+        print(request.validated_data)
```
Quitar los prints, solo los usamos en desarrollo para validar los valores de las variables

cla1976 commented on user_profile_api/views.py:
```diff
@@ -73,11 +73,17 @@ def post(self, request):
             if res.status_code >= 300:
                 raise ValueError('Error perform record!')
             else:                
+                print("Hola test", res)
+                print('Esto es', serializer.validated_data)
+                begin_time = serializer.validated_data["begin_time"]
+                begin_time_final = begin_time.strftime("%Y-%m-%dT%H:%M:%S")
+                print(begin_time_final)
+                      
```
Quitar los prints

cla1976 commented on user_profile_api/views.py:
```diff
@@ -86,8 +92,8 @@ def post(self, request):
     serializer_class = serializers.UserProfileSerializer
-    #authentication_classes = (TokenAuthentication,)
-    #permission_classes = (permissions.UpdateOwnProfile,)
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (permissions.UpdateOwnProfile,)
```
Dejar comentada estas 2 lineas

cla1976 commented on users_admin/settings.py:
```diff
@@ -59,7 +59,7 @@
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [BASE_DIR / 'templates'],
+        'DIRS': [],
```
Dejar esta linea como esta no modificarla

cla1976 commented on users_admin/settings.py:
```diff
@@ -123,11 +123,6 @@
 
 STATIC_URL = 'static/'
 
-STATICFILES_DIRS = [
-    BASE_DIR/'static'
-]
-
-
```
Dejar este codigo tal como esta

cla1976 commented on users_admin/settings.py:
```diff
@@ -136,7 +131,7 @@
 
 # Variable to identify environment 'TEST' or 'PROD' (if 'TEST' use mocked requests)
-ENVIRONMENT = 'TEST'
+ENVIRONMENT = 'PROD'
```
Dejar el environment siempre en TEST

cla1976 reviewed (changes requested):
Revisado, hacer todos los cambios solicitados

Commit:
Se realizan los cambios necesarios.

j2c20001 committed:
Cambios realizados mayormente con respecto a la base de datos, carga de
usuarios y diseño interno.
Se finaliza el ABM, junto a la carga de imágenes al dispositivo.
Cambios marcados por archivos:

-admin.py
Se configura cada administrador de los modelos para cargar las
filas en las tablas. Se determinan los diferentes campos disponibles,
como los filtros y demás aspectos en la forma que se mostrará en
pantalla los campos para cargar en la
base de datos.

-mocks.py y serializers.py
Se reemplaza el diseño anterior de Django REST Framework
para subir los usuarios con el archivo de signals.py que en base
al panel de administrador de Django utiliza el concepto de señales
para cargar, eliminar y modificar las diferentes filas de los
modelos.

-signals.py
Se configuran las señales que actuarán en base a una acción
en el panel de administrador de Django tales como pre_save, post_save,
pre_delete, post_delete.
En cada una de estas señales se envía un JSON con su función respectiva
al dispositivo.
Cambiando la variable del mockeo a verdadero puede utilizarse la
carga de usuarios sin necesidad de estar conectado al dispositivo.

-models.py
Se agregan los campos respectivos para que correlacionen con
la carga de datos al dispositivo y algunas tablas que pueden ser
consideradas útiles en un futuro.

-services.py
Se quita lo relacionado con los archivos reemplazados y se coloca
una función que sirve para determinar el valor único e irrepetible
de la ID principal de cada usuario en base a la base de datos.
Es posible que luego se agregue la funcionalidad de que también
se revise en base a la base de datos y en base al dispositivo
en que se pretende cargar el usuario.

-views.py
Por ahora no se modifica nada de las vistas ya que con lo proporcionado
por el panel de administración predeterminado de Django se
consiguieron ciertas funcionalidades.

j2c20001 committed:
Merge pull request #9 from j2c20001/feature/new-django-backend

Diseño backend

j2c20001 committed:
Fix

cla1976 reviewed (dismissed):
Ok!

Commit:
Agregado frontend base

j2c20001 committed:
Merge pull request #10 from cuervocoder/new-users-json-front

Se agrega stream de cámara
